### ISSUE TITLE
The problem occurs durring the training

### ISSUE THREAD

**Issue by huazai-1994** [wolny/pytorch-3dunet#27]
2019-06-11 17:56:42,646 [MainThread] INFO UNet3DTrainer - Validation iteration 188
2019-06-11 17:56:42,768 [MainThread] INFO UNet3DTrainer - Validation iteration 189
2019-06-11 17:56:42,887 [MainThread] INFO UNet3DTrainer - Validation iteration 190
2019-06-11 17:56:43,007 [MainThread] INFO UNet3DTrainer - Validation iteration 191
Warning: NaN or Inf found in input tensor.
2019-06-11 17:56:43,259 [MainThread] INFO UNet3DTrainer - Validation finished. Loss: 0.33309157844632864. Evaluation score: nan
2019-06-11 17:56:43,262 [MainThread] INFO UNet3DTrainer - Saving last checkpoint to '3dunet/last_checkpoint.pytorch'
2019-06-11 17:56:43,314 [MainThread] INFO UNet3DTrainer - Training stats. Loss: 0.0765402484126389. Evaluation score: nan
Warning: NaN or Inf found in input tensor.
2019-06-11 17:56:43,315 [MainThread] INFO UNet3DTrainer - Logging model parameters and gradients
2019-06-11 17:56:44,325 [MainThread] INFO UNet3DTrainer - Training iteration 101. Batch 100. Epoch [0/49]
2019-06-11 17:56:44,464 [MainThread] INFO UNet3DTrainer - Training iteration 102. Batch 101. Epoch [0/49]
2019-06-11 17:56:44,913 [MainThread] INFO UNet3DTrainer - Training iteration 103. Batch 102. Epoch [0/49]
#############
As listed above, there is a warning , and the problem :'Evaluation score:nan' .So could you please give me some help to solve the problem ? I guess maybe there is something with my own data ,but i have no idea how to fix it Thanks a lot

**Comment by StuckinPhD** [comment]
> Warning: NaN or Inf found in input tensor.

You dont have any input tensor - Check your data loader or debug the hdf5.py file to see why is your Input Tensor NaN.

**Comment by wolny** [comment]
Hi @huazai-1994,

what validation metric are you using during training? It might be that my implementation of that metric doesn't work with your data. I'd follow the suggestion from @StuckinPhD and try to debug the function that computes the metric during training and see why it produces a Nan.

**Comment by huazai-1994** [comment]
@wolny hi thanks for your reply! I have used the MeanIoU as the metric. I have checked my own dataset,but found noting wrong.(Ps:the scales of my original datas are uncertain,eg 128*1056*892,123*1133*1144,so before saved as h5 file, I have cropped or padded with 0 that make the data to a certrain scale eg 128*1024*1024 123*1024*1024 but the depth is not changed)

**Comment by antonioricciardi16** [comment]
The bug is in the code, not your data.

See the last lines of trainer.py:

return (img - np.min(img)) / np.ptp(img)

This can easily be 0/0, which causes the function to return NaN. I suggest you fix it by simply adding an epsilon to the denominator, so that the function outputs 0 rather than NaN:

import sys
return (img - np.min(img)) / (np.ptp(img) + sys.float_info.epsilon)

Same for metrics.py, line 109:

return torch.sum(prediction & target).float() / (torch.sum(prediction | target).float() + sys.float_info.epsilon)

**Comment by sjyun23** [comment]
@antonioricciardi16 it really works!! thank you:)
